src/Migrator/MigrationsTable.php

```diff
@@ -0,0 +1,166 @@
+     */
+    private const MIGRATION_TABLE_INDICES_LIST = [
+        'migration',
+        'created_at'
+    ];
+
+    /**
+     * @var Database
+     */
+    private $db;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var AbstractTable
+     */
+    private $schema;
+
+    /**
+     * @param Database $db
+     * @param string $name
+     */
+    public function __construct(Database $db, string $name)
+    {
+        $this->db = $db;
+        $this->name = $name;
+
+        $table = $db->table($name);
+        $this->schema = $table->getSchema();
+    }
+
+    /**
+     * Schema update will automatically sync all needed data.
+     *
+     * Please note that if you change this migration, you will also need to
+     * change the list of fields in this migration specified in
+     * the {@see MigrationsTable::MIGRATION_TABLE_FIELDS_LIST}
+     * and {@see MigrationsTable::MIGRATION_TABLE_INDICES_LIST} constants.
+     *
+     * @return void
+     */
+    public function actualize(): void
+    {
+        $this->schema->primary('id');
+        $this->schema->string('migration', 191)->nullable(false);
+        $this->schema->datetime('time_executed')->datetime();
+        $this->schema->datetime('created_at')->datetime();
+
+        $this->schema->index(['migration', 'created_at'])
+            ->unique(true);
+
+        if ($this->schema->hasIndex(['migration'])) {
+            $this->schema->dropIndex(['migration']);
+        }
+
+        $this->schema->save();
+    }
+
+    /**
+     * Returns {@see true} if the migration table in the database is up to date
+     * or {@see false} instead.
+     *
+     * @return bool
+     */
+    public function isPresent(): bool
+    {
+        if (!$this->isTableExists()) {
+            return false;
+        }
+
+        if (!$this->isNecessaryColumnsExists()) {
+            return false;
+        }
+
+        if (!$this->isNecessaryIndicesExists()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns {@see true} if the migration's table exists or {@see false}
+     * instead.
+     *
+     * @return bool
+     */
+    private function isTableExists(): bool
+    {
+        return $this->db->hasTable($this->name);
+    }
+
+    /**
+     * Returns {@see true} if all migration's fields is present or {@see false}
+     * otherwise.
+     *
+     * @return bool
+     */
+    private function isNecessaryColumnsExists(): bool
+    {
+        foreach (self::MIGRATION_TABLE_FIELDS_LIST as $field) {
+            if (!$this->schema->hasColumn($field)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns {@see true} if all migration's indices is present or {@see false}
+     * otherwise.
+     *
+     * @return bool
+     */
+    private function isNecessaryIndicesExists(): bool
+    {
+        return $this->schema->hasIndex(self::MIGRATION_TABLE_INDICES_LIST);
```

Comment:
I don't sure that check is correct. Is this index composite? If not then i think you should use `foreach` here

Reply:
See `actualize()` on line 95